pkg/helpers/sa_syncer.go

```diff
@@ -0,0 +1,92 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// EnsureSAToken get the saToken of target sa and ensure it was rendered as expected.
+// A usage example combined with RenderToKubeconfigSecert would be like the following:
+// ...
+// saName := clusterManagerName+"-registration-controller-sa"
+// secretName := saName+"-kubeconfig"
+// err := helpers.EnsureSAToken(ctx, cluster, saName, clustermanagerNamespace, hostedAmdinKubeClient,
+// 	helpers.RenderToKubeconfigSecret(secretName, clustermanagerNamespace, hostedAdminKubeconfig, n.kubeClient, n.recorder))
```

Comment:
Could we change the `n.kubeClient` and `n.recorder` since we don't know what is `n`?

Reply:
Done